overlord/snapshotstate/backend/helpers.go

```diff
@@ -152,26 +153,53 @@ func allUsers() ([]*user.User, error) {
 	return users, nil
 }
 
+var (
+	sysGeteuid   = sys.Geteuid
+	execLookPath = exec.LookPath
+)
+
+func pickUserWrapper() string {
+	// runuser and sudo happen to work the same way in this case.
+	// The main reason to prefer runuser over sudo is that runuser
+	// is part of util-linux, which is considered essential,
+	// whereas sudo is an addon which could be removed.
+	for _, cmd := range []string{"runuser", "sudo"} {
+		if lp, err := execLookPath(cmd); err == nil {
+			return lp
+		}
+	}
+	return ""
+}
+
+var sudo = pickUserWrapper()
+
 // maybeRunuserCommand returns an exec.Cmd that will, if the current
 // effective user id is 0 and username is not "root", call runuser(1)
-// to change to the given username before running the given command.
+// (or sudo(8) if runuser(1) isn't available) to change to the given
+// username before running the given command.
 //
 // If username is "root", or the effective user id is 0, the given
 // command is passed directly to exec.Command.
 //
 // TODO: maybe move this to osutil
 func maybeRunuserCommand(username string, args ...string) *exec.Cmd {
-	if username == "root" || sys.Geteuid() != 0 {
-		// runuser only works for euid 0, and doesn't make sense for root
-		return exec.Command(args[0], args[1:]...)
+	if sysGeteuid() == 0 && username != "root" && sudo != "" {
+		sudoArgs := make([]string, len(args)+4)
+		copy(sudoArgs[4:], args)
+		sudoArgs[0] = sudo
+		sudoArgs[1] = "-u"
+		sudoArgs[2] = username
+		sudoArgs[3] = "--"
+		return &exec.Cmd{
+			Path: sudo,
+			Args: sudoArgs,
+		}
 	}
-	sudoArgs := make([]string, len(args)+3)
-	copy(sudoArgs[3:], args)
-	sudoArgs[0] = "-u"
-	sudoArgs[1] = username
-	sudoArgs[2] = "--"
 
-	return exec.Command("runuser", sudoArgs...)
+	// TODO: use warnings instead
+	logger.Noticef("No user wrapper found; running tar for user data as root. Please make sure 'sudo' or 'runuser' (from util-linux) is on $PATH to avoid this.")
```

Comment:
Hmm, when this is the case what would happen? Should things just fail loudly?

Reply:
I'm not sure. It'll break on NFS and you'll get an error. Other than that it should be fine.

Reply:
Ah, that's a good thing to know. We can already check if home is using NFS so perhaps all of this can be optional eventually?